action/rlp_tx_test.go

```diff
@@ -390,18 +395,36 @@ func TestIssue3944(t *testing.T) {
 		S:        new(big.Int).SetBytes(sig[32:64]),
 	})
 
+	r.EqualValues(types.LegacyTxType, tx.Type())
+	r.False(tx.Protected())
 	v, q, s := tx.RawSignatureValues()
 	r.Equal(sig[:32], q.Bytes())
 	r.Equal(sig[32:64], s.Bytes())
 	r.Equal("1b", v.Text(16))
 	r.NotEqual(hash, tx.Hash().Hex()) // hash does not match with wrong V value in signature
 
-	signer := types.NewEIP155Signer(big.NewInt(4690))
+	signer, err := EthSigner(iotextypes.Encoding_ETHEREUM_EIP155, 4690)
+	r.NoError(err)
 	tx1, err := RawTxToSignedTx(tx, signer, sig)
 	r.NoError(err)
+	r.True(tx1.Protected())
+	r.EqualValues(4690, tx1.ChainId().Uint64())
 	v, q, s = tx1.RawSignatureValues()
 	r.Equal(sig[:32], q.Bytes())
 	r.Equal(sig[32:64], s.Bytes())
 	r.Equal("9415", v.String()) // this is the correct V value corresponding to chainID = 4690
 	r.Equal(hash, tx1.Hash().Hex())
 }
+
+func TestDetermine(t *testing.T) {
+	r := require.New(t)
+
+	tx, encoding, sig, pubkey, err := DecodeEtherTx("0xf8a58085174876e800830186a08080b853604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf31ba02222222222222222222222222222222222222222222222222222222222222222a02222222222222222222222222222222222222222222222222222222222222222")
```

Comment:
unit test of `DecodeEtherTx` never covers the case of `err != nil`

Reply:
right, I am working to add more tests